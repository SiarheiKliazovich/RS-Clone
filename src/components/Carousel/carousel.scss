@import "../../sass/variables";
@import "../../sass/mixins";


.section-testimonial {
  background: #4788ff;
  padding: 30px 0;
}

.section-testimonial-wrapper {
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}

.section-testimonial-title {
  display: flex;
  margin-bottom: 60px;
  justify-content: space-between;
  flex-direction: column;
  align-items: center;
}


.title-design {
  font-weight: 400;
  color: #ffffff;
  display: inline-block;
  position: relative;
  text-transform: uppercase;
  padding-bottom: 10px;
  font-family: $font-family-base;
}

.testimonial-title {
  color: #ffffff;
  font-family: $font-family-base;
  font-size: $font-size-xxlg;
}


.carousel {

}
// ==============================================logic


.slideshow {
  margin: 0 auto;
  overflow: hidden;
  max-width: 500px;
}

.slideshowSlider {
  white-space: nowrap;
  transition: ease 1000ms;
}

.slide {
  display: inline-block;
  width: 100%;
 
}

/* Buttons */

.slideshowDots {
  text-align: center;
  display: flex;
  justify-content: center;
}


.slideshowDot {
  display: inline-block;
  height: 20px;
  width: 20px;
  border-radius: 50%;

  cursor: pointer;
  margin: 15px 7px 0px;
  background-color: #c4c4c4;
}

.slideshowDot.active {
  background-color: #6a0dad;
}


// ============================================
.testimonial-item {
  
  box-shadow: 4.871px 34.659px 30px 0 rgb(0 0 0 / 5%);
  background: #fff;
  border-radius: 5px;
  white-space:normal;

   &:first-child {
    margin-right: 100px;
   }
   &:last-child {
    margin-left: 100px;
   }
}

.user-wrapper {
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding-top: 10px;
}

.user-info {
  padding-left: 20px;
}

.name-user {
  font-family: $font-family-base;
  font-size: $font-size-xlg;
  color: #000000;
}

.name-user-job {
  font-family: $font-family-base;
}

.icon-one {
  background-image: url(../../assets/images/jpg/avarat1.jpg);
  background-repeat: no-repeat;
  width: 37px;
  height: 37px;
}

.icon-two {
  background-image: url(../../assets/images/jpg/avatar10.jpg);
  background-repeat: no-repeat;
  width: 37px;
  height: 37px;
}

.icon-tree {
  background-image: url(../../assets/images/jpg/avatar9.jpg);
  background-repeat: no-repeat;
  width: 37px;
  height: 37px;
}

.testimonial-block {
  padding: 20px 45px;
}

.testimonial-text {
  font-family: $font-family-base;
  font-size: $font-size-llg;
}