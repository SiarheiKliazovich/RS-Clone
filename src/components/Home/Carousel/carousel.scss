@import "../../../sass/variables";
@import "../../../sass/mixins";


.section-testimonial {
  background: $color-white1;
  padding: 30px 0;
}

.section-testimonial-wrapper {
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}

.section-testimonial-title {
  display: flex;
  margin-bottom: 60px;
  justify-content: space-between;
  flex-direction: column;
  align-items: center;
  text-align: center;
}

.title-design {
  font-weight: $font-weight-normal;
  color: $color-grey7;
  display: inline-block;
  position: relative;
  text-transform: uppercase;
  padding-bottom: 10px;
  font-family: $font-family-base;
}

.testimonial-title {
  color: $color-grey7;
  font-family: $font-family-base;
  font-size: $font-size-xxlg;
}

// ==============================================logic

.slideshow {
  margin: 0 auto;
  overflow: hidden;
  width: 100%;
}

.slideshowSlider {
  white-space: nowrap;
  transition: ease 1000ms;
}

/*============================= Buttons */

.slideshowDots {
  text-align: center;
  display: flex;
  justify-content: center;
}

.slideshowDot {
  display: inline-block;
  height: 20px;
  width: 20px;
  border-radius: 50%;
  margin: 15px 7px 0px;
  background-color: $color-grey1;
  cursor: pointer;
  z-index: 9;
}


// ============================================
.slide {
  display: inline-block;
  width: 94%;
}

.testimonial-item {
  box-shadow: 4.871px 34.659px 30px 0 $color-black3;
  background: $color-white2;
  border-radius: 5px; 
  white-space:normal;
  margin: 0 35px 40px 35px;

  @include media-small-tablet {
    margin: 0 25px 40px 25px;
  }

  @include media-mobile {
    margin: 0 0px 40px 35px;
  }

  @include media-small-mobile {
    margin:0px 13px 40px 15px;
  }
}

.testimonial-item__wrapper {
  text-align: center;
  padding: 15px 15px;
}

.user-wrapper {
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding-top: 10px;
}

.user-img-carousel {
  width: 37px;
  height: 37px;
}

.user-info {
  padding-left: 20px;
}

.name-user {
  font-family: $font-family-base;
  font-size: $font-size-xlg;
  color: $color-black1;
}

.name-user-job {
  font-family: $font-family-base;
}

.testimonial-text {
  font-family: $font-family-base;
  font-size: $font-size-llg;
  padding: 20px 45px;
}