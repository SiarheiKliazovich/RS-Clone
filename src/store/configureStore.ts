/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
