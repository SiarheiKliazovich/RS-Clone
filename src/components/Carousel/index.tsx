import { FunctionComponent, useEffect, useState, useRef } from "react";
import "./carousel.scss";
import { СarouselType } from "../../types/home";
import { nightTheme } from "../../data/constants";

const Сarousel: FunctionComponent<СarouselType> = ({
  accentColor,
  isNavbarNightMode,
}: СarouselType) => {
  const testimonial = ["1", "2", "3"];
  const delay = 2500;
  const [index, setIndex] = useState(0);
  const timeoutRef = useRef(null);
  const backgroundColor = nightTheme.background.element;

  const resetTimeout = () => {
    if (timeoutRef.current) {
      clearTimeout(timeoutRef.current);
    }
  };

  useEffect(() => {
    resetTimeout();
    timeoutRef.current = setTimeout(
      () =>
        setIndex((prevIndex) =>
          prevIndex === testimonial.length - 1 ? 0 : prevIndex + 1
        ),
      delay
    );
    return () => {
      resetTimeout();
    };
  }, [index]);

  return (
    <div
      className="section-testimonial"
      style={{
        backgroundColor: isNavbarNightMode ? backgroundColor : accentColor,
      }}
    >
      <div className="section-testimonial-wrapper">
        <div className="section-testimonial-title">
          <span className="title-design">TESTIMONIALS</span>
          <h2 className="testimonial-title">
            +100500 Companies and People who have used TieConfigurator
          </h2>
        </div>
        <div className="carousel">
          <div className="slideshow">
            <div className="carousel-wrapper">
              <div
                className=" slideshowSlider"
                style={{ transform: `translate3d(${-index * 120}%, 0, 0)` }}
              >
                <div className="slide testimonial-item">
                  <div className="user-wrapper">
                    <div className="icon-one"></div>
                    <div className="user-info">
                      <p className="name-user">John Malcovich</p>
                      <p className="name-user-job">Chimney Sweep On Call</p>
                    </div>
                  </div>
                  <div className="testimonial-block">
                    <p className="testimonial-text">
                      Hi--The tie I just purchased is for a young friend's
                      January birthday. The black tie I bought him last year
                      with the musical notes was a first-rate home run, so I
                      have every expectation that this year's blue tie with the
                      la crosse pattern will be another smashing success! You
                      guys offer a terrific variety from which to purchase for
                      my discerning young friend, and I look forward to making
                      annual purchases from you for many years to come.
                    </p>
                  </div>
                </div>
                <div className="slide testimonial-item">
                  <div className="user-wrapper">
                    <div className="icon-two"></div>
                    <div className="user-info">
                      <p className="name-user">Lisa Jeffrey</p>
                      <p className="name-user-job">Manager</p>
                    </div>
                  </div>
                  <div className="testimonial-block">
                    <p className="testimonial-text">
                      Thank you, ConfiguratorTies.com, for being amazing!I
                      placed my second order with this website for a tie for my
                      boyfriend to match my dress, and both have matched
                      perfectly! This website has the best, most affordable,
                      most diverse selection of ties with excellent customer
                      service. I'll never buy a tie anywhere else.
                    </p>
                  </div>
                </div>
                <div className="slide testimonial-item">
                  <div className="user-wrapper">
                    <div className="icon-tree"></div>
                    <div className="user-info">
                      <p className="name-user">Billy Bones</p>
                      <p className="name-user-job">Donut Baker</p>
                    </div>
                  </div>
                  <div className="testimonial-block">
                    <p className="testimonial-text">
                      I love your company and love your great ties. The person
                      on the phone was so helpful and courteous. The tie arrived
                      so quickly and was wonderful. In this day and age, it's so
                      nice to see such dedication and hard work to create and
                      sustain a great company. Thanks!
                    </p>
                  </div>
                </div>
              </div>

              <div className="slideshowDots">
                {testimonial.map((_, idx) => (
                  <div
                    key={idx}
                    className={`slideshowDot${index === idx ? " active" : ""}`}
                    onClick={() => {
                      setIndex(idx);
                    }}
                  ></div>
                ))}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
export default Сarousel;
