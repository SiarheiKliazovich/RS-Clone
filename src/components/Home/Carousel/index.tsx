import { FunctionComponent, useEffect, useState, useRef } from "react";
import "./carousel.scss";
import { СarouselType } from "../../../types/home";
import { nightTheme } from "../../../data/constants";
import carouselLang from "../../../data/carousel";

const Сarousel: FunctionComponent<СarouselType> = ({
  accentColor,
  isNavbarNightMode,
}: СarouselType) => {
  const testimonial = ["1", "2", "3"];
  const delay = 2500;
  const [index, setIndex] = useState(0);
  const timeoutRef = useRef(null);
  const backgroundColor = nightTheme.background.element;

  const resetTimeout = () => {
    if (timeoutRef.current) {
      clearTimeout(timeoutRef.current);
    }
  };

  useEffect(() => {
    resetTimeout();
    /* eslint-disable @typescript-eslint/ban-ts-comment */
    // @ts-ignore
    timeoutRef.current = setTimeout(
      () =>
        setIndex((prevIndex) =>
          prevIndex === testimonial.length - 1 ? 0 : prevIndex + 1
        ),
      delay
    );
    /* eslint-enable */
    return () => {
      resetTimeout();
    };
  }, [index]);
  const lang = "ru";
  const list = carouselLang.find((c) => c.lang === lang)!;

  return (
    <div className="section-testimonial">
      <div className="container">
        <div className="section-testimonial-title">
          <span className="title-design">{list.title.littletitle}</span>
          <h2 className="testimonial-title">{list.title.bigtitle}</h2>
        </div>
        <div className="carousel">
          <div className="slideshow">
            <div className="carousel-wrapper">
              <div
                className=" slideshowSlider"
                style={{ transform: `translate3d(${-index * 100}%, 0, 0)` }}
              >
                {list.data.map((item) => (
                  <div className="slide testimonial-item" key={item.nameUser}>
                    <div className="testimonial-item__wrapper">
                      <div className="user-wrapper">
                        <img
                          className="user-img-carousel"
                          src={item.image}
                          alt=""
                        />
                        <div className="user-info">
                          <p className="name-user">{item.nameUser}</p>
                          <p className="name-user-job">{item.profUser}</p>
                        </div>
                      </div>
                      <p className="testimonial-text">{item.testimonials}</p>
                    </div>
                  </div>
                ))}
              </div>
              <div className="slideshowDots">
                {testimonial.map((_, idx) => (
                  <div
                    key={idx}
                    className="slideshowDot"
                    style={{
                      backgroundColor: isNavbarNightMode
                        ? backgroundColor
                        : index === idx
                        ? accentColor.static
                        : "#959393",
                    }}
                    onClick={() => {
                      setIndex(idx);
                    }}
                  ></div>
                ))}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
export default Сarousel;
