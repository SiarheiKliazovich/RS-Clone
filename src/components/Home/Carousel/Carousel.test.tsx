import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import Carousel from ".";

const mockStore = configureStore([]);

describe("Carousel", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      langReducer: {
        lang: "en",
      },
    });
  });

  test("renders correctly", () => {
    const accentColor = {
      static: "#000000",
      hover: "#ffffff",
    };
    const isNavbarNightMode = false;

    const { container } = render(
      <Provider store={store}>
        <Router>
          <Carousel
            accentColor={accentColor}
            isNavbarNightMode={isNavbarNightMode}
          />
        </Router>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
