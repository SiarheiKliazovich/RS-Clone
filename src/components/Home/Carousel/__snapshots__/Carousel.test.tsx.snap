// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders correctly 1`] = `
<div>
  <div
    class="section-testimonial"
  >
    <div
      class="container"
    >
      <div
        class="section-testimonial-title"
      >
        <span
          class="title-design"
        >
          TESTIMONIALS
        </span>
        <h2
          class="testimonial-title"
        >
          +100500 Companies And People Who Have Used TieConfigurator
        </h2>
      </div>
      <div
        class="carousel"
      >
        <div
          class="slideshow"
        >
          <div
            class="carousel-wrapper"
          >
            <div
              class=" slideshowSlider"
              style="transform: translate3d(0%, 0, 0);"
            >
              <div
                class="slide testimonial-item"
              >
                <div
                  class="testimonial-item__wrapper"
                >
                  <div
                    class="user-wrapper"
                  >
                    <img
                      alt=""
                      class="user-img-carousel"
                      src="https://i.ibb.co/28gV9DC/avarat-11-1.jpg"
                    />
                    <div
                      class="user-info"
                    >
                      <p
                        class="name-user"
                      >
                        John Malcovich
                      </p>
                      <p
                        class="name-user-job"
                      >
                        Chimney Sweep On Call
                      </p>
                    </div>
                  </div>
                  <p
                    class="testimonial-text"
                  >
                    Hi--The tie I just purchased is for a young friend's January birthday. The black tie I bought him last year with the musical notes was a first-rate home run, so I have every expectation that this year's blue tie with the la crosse pattern will be another smashing success!
                  </p>
                </div>
              </div>
              <div
                class="slide testimonial-item"
              >
                <div
                  class="testimonial-item__wrapper"
                >
                  <div
                    class="user-wrapper"
                  >
                    <img
                      alt=""
                      class="user-img-carousel"
                      src="https://i.ibb.co/pv7kxhK/avatar-10-1.jpg"
                    />
                    <div
                      class="user-info"
                    >
                      <p
                        class="name-user"
                      >
                        Lisa Jeffrey
                      </p>
                      <p
                        class="name-user-job"
                      >
                        Manager
                      </p>
                    </div>
                  </div>
                  <p
                    class="testimonial-text"
                  >
                    Thank you,TiesCreator.com,for being amazing!I placed my second order with this website for a tie for my BF to match my dress, and both have matched perfectly! This website has the best, most affordable, most diverse selection of ties with excellent customer service.
                  </p>
                </div>
              </div>
              <div
                class="slide testimonial-item"
              >
                <div
                  class="testimonial-item__wrapper"
                >
                  <div
                    class="user-wrapper"
                  >
                    <img
                      alt=""
                      class="user-img-carousel"
                      src="https://i.ibb.co/XL2wpfL/avatar-9-1.jpg"
                    />
                    <div
                      class="user-info"
                    >
                      <p
                        class="name-user"
                      >
                        Billy Bones
                      </p>
                      <p
                        class="name-user-job"
                      >
                        Donut Baker
                      </p>
                    </div>
                  </div>
                  <p
                    class="testimonial-text"
                  >
                    I love your company and love your great ties. The person on the phone was so helpful and courteous. The tie arrived so quickly and was wonderful. In this day and age, it's so nice to see such dedication and hard work to create and sustain a great company. Thanks!
                  </p>
                </div>
              </div>
            </div>
            <div
              class="slideshowDots"
            >
              <div
                class="slideshowDot"
                style="background-color: rgb(0, 0, 0);"
              />
              <div
                class="slideshowDot"
                style="background-color: rgb(149, 147, 147);"
              />
              <div
                class="slideshowDot"
                style="background-color: rgb(149, 147, 147);"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
